### PR TITLE
starting Block storage work : plumbing "volume" in jumpgate 

### PR DESCRIPTION
**Pull request by Neetuj**
-changing block storage to volume so that load_endpoints can pick volume endpoints
-enabling volume in config file
-mapping volume/detail endpoint to empty function .. build on this more .. 

result 

REQ: curl -i 'http://localhost:5000/volume/v2/13193/volumes/detail' -X GET -H "User-Agent: python-novaclient" -H "Accept: application/json" -H "X-Auth-Token: SDFhOTZNek9GY1kwKzkyZm9GUUREd1MzcGhnaGhqZjUyTUN2d3Z2eC90NVNUT1J5bHFDNEJ4QmpuWS8xTndRZWNmR1ZhUENIbnJHRlhTTGVTSW1JWENrSTg4cWhWTTFGQWhxN01IYmtJQ0lIZ0lwMk5pVys4MGRBd1RVYmtpVjlCWXhhL2Q2b0ZQU1RnT0k1UlRuVFJRNi9raHB6bXJBN0dtQUdESU04VWU4ZDRnWkc2bk81d0tNampMOG5QQThqVjFFTmpSejVKd29jQVFWbDNaZFZDNHF2Wkc5cm5BRDVFdVMySDdEOE9XLzlITDB2aTUyVjUwYTlnSWZ3eERuNQ=="

RESP: [200] CaseInsensitiveDict({'content-length': '15', 'x-compute-request-id': 'req-11b3c0c8-dd17-11e3-8a3e-064923b0a07b', 'server': 'gunicorn/18.0', 'connection': 'close', 'date': 'Fri, 16 May 2014 16:28:10 GMT', 'content-type': 'application/json'})
RESP BODY: {"volumes": []}

+----+--------+--------------+------+-------------+-------------+
| ID | Status | Display Name | Size | Volume Type | Attached to |
+----+--------+--------------+------+-------------+-------------+
+----+--------+--------------+------+-------------+-------------+

### CODE REVIEW

**Comment by sudorandom** [comment]
@Neetuj the build failed due to pylint errors: https://travis-ci.org/softlayer/jumpgate/builds/25346234

File: jumpgate/volume/drivers/sl/volumes.py [hunk 1]
```diff
@@ -1,8 +1,28 @@
-
+from SoftLayer import ISCSIManager
 
 class VolumesV2(object):
+
+
+
     def on_get(self, req, resp, tenant_id):
-        resp.body = {'volumes': []}
```
Comment (sudorandom) [inline jumpgate/volume/drivers/sl/volumes.py:5]: Could you revert this file until some of the functionality is completed?

Comment (Neetuj) [inline jumpgate/volume/drivers/sl/volumes.py:5]: ok

**Comment by Neetuj** [comment]
added support for nova volume list .. right now looks at iscsimanager and lists what you will see in "sl iscsi list"  

RESP: [200] CaseInsensitiveDict({'content-length': '1478', 'x-compute-request-id': 'req-5a02aac4-e050-11e3-8a3e-064923b0a07b', 'server': 'gunicorn/18.0', 'connection': 'close', 'date': 'Tue, 20 May 2014 18:55:47 GMT', 'content-type': 'application/json'})
RESP BODY: {"volumes": [{"display_name": "storage501", "attachments": [], "created_at": "2012-12-19T16:33:44-06:00", "volume_type": "ISCSI", "id": 1575636, "size": 20}, {"display_name": "storage501", "attachments": [], "created_at": "2012-12-20T09:07:30-06:00", "volume_type": "ISCSI", "id": 1576574, "size": 20}, {"display_name": "storage501", "attachments": [], "created_at": "2013-04-12T12:25:33-06:00", "volume_type": "ISCSI", "id": 1751318, "size": 500}, {"display_name": "storage501", "attachments": [], "created_at": "2013-06-05T07:48:11-06:00", "volume_type": "ISCSI", "id": 1857418, "size": 120}, {"display_name": "storage301", "attachments": [], "created_at": "2013-08-22T14:28:39-06:00", "volume_type": "ISCSI", "id": 2020151, "size": 100}, {"display_name": "storage151", "attachments": [], "created_at": "2013-09-05T16:04:09-06:00", "volume_type": "ISCSI", "id": 2057920, "size": 40}, {"display_name": "storageams0101", "attachments": [], "created_at": "2013-10-27T09:04:39-06:00", "volume_type": "ISCSI", "id": 2152728, "size": 40}, {"display_name": "storage201", "attachments": [], "created_at": "2013-11-06T01:58:34-06:00", "volume_type": "ISCSI", "id": 2160610, "size": 500}, {"display_name": "storagedal0103", "attachments": [], "created_at": "2013-12-19T18:42:32-06:00", "volume_type": "ISCSI", "id": 2226679, "size": 1}, {"display_name": "storagedal0506", "attachments": [], "created_at": "2014-02-27T14:55:15-06:00", "volume_type": "ISCSI", "id": 2531984, "size": 20}]}

+---------+--------+----------------+------+-------------+-------------+
| ID      | Status | Display Name   | Size | Volume Type | Attached to |
+---------+--------+----------------+------+-------------+-------------+
| 1575636 |        | storage501     | 20   | ISCSI       |             |
| 1576574 |        | storage501     | 20   | ISCSI       |             |
| 1751318 |        | storage501     | 500  | ISCSI       |             |
| 1857418 |        | storage501     | 120  | ISCSI       |             |
| 2020151 |        | storage301     | 100  | ISCSI       |             |
| 2057920 |        | storage151     | 40   | ISCSI       |             |
| 2152728 |        | storageams0101 | 40   | ISCSI       |             |
| 2160610 |        | storage201     | 500  | ISCSI       |             |
| 2226679 |        | storagedal0103 | 1    | ISCSI       |             |
| 2531984 |        | storagedal0506 | 20   | ISCSI       |             |
+---------+--------+----------------+------+-------------+-------------+

**Comment by mjfork** [comment]
Wouldn't a more accurate match for the OpenStack volume service be the Portable Storage offering, not iSCSI volumes.  Portable Storage appears in a VM as an additional attached HDD vs iSCSI requiring work in the guest to mount.

**Comment by Neetuj** [comment]
@mjfork will look into it .. currently soflayer has nas and iscsi  listed in storage so thought of started with that ...

**Comment by Neetuj** [comment]
once this change is picked up build should not fail 
https://github.com/softlayer/softlayer-python/blob/master/SoftLayer/managers/iscsi.py#L71

**Comment by sudorandom** [comment]
Closing in favor of portable storage volumes over iSCSI-based volumes.

**Pull request closed without merging**